lib/seadsa/DsaGlobal.cc

```diff
@@ -694,7 +694,8 @@ bool BottomUpGlobalAnalysis::runOnModule(Module &M) {
   LOG("dsa-global", errs() << "Started bottom-up global analysis ... \n");
 
   for (auto &F : M) {
-    if (F.isDeclaration() || F.empty()) continue;
+    if (F.empty() && !m_specGraphInfo.hasSpecFunc(F)) continue;
```

Comment:
pull the common condition out

```
if (!m_specGraphInfo.hasSpecFunc(F)) {
     if (F.empty() || F.isDeclaration()) continue;
}
```

Reply:
I don't think this is the same logic. If the function is defined but and don't have a spec for it my code won't `continue` but yours will